lib/logthrdestdrv.c

```diff
@@ -555,70 +559,292 @@ _init_watches(LogThreadedDestDriver *self)
 }
 
 static void
-_signal_startup_finished(LogThreadedDestDriver *self)
+_signal_startup_finished(LogThreadedDestWorker *self, gboolean thread_failure)
 {
-  g_mutex_lock(self->lock);
+  g_mutex_lock(self->owner->lock);
   self->startup_finished = TRUE;
+  self->startup_failure |= thread_failure;
```

Comment:
> We want to fail if any of the workers failed.
My point was that `startup_failure` is a Worker attribute and not the Driver's, therefore I don't think it can be evaluated multiple times, hence .
But it was just a question, it caught my attention, it looked interesting. :)